libglusterfs/src/inode.c

```diff
@@ -1651,17 +1701,18 @@ inode_table_prune(inode_table_t *table)
     /* Pick 1 inode for invalidation */
     if (tmp) {
         xlator_t *old_THIS = THIS;
+        gf_boolean_t tmp_is_root = __is_root_gfid(tmp->gfid);
         THIS = table->invalidator_xl;
         ret1 = table->invalidator_fn(table->invalidator_xl, tmp);
         THIS = old_THIS;
         pthread_mutex_lock(&table->lock);
         {
             if (!ret1) {
                 tmp->invalidate_sent = true;
-                __inode_unref(tmp, false);
+                (void)(!tmp_is_root && __inode_unref(tmp, false, old_THIS));
```

Comment:
Bad way of writing code. Please use a regular `if` statement.

Reply:
Noted and addressed.